### PR TITLE
feat: include action references in layout service

### PR DESCRIPTION
**Pull request by diego-transferwise**
## 🚀 Changes
Now layout service can inline the layout of action references

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.js [hunk 1]
```diff
@@ -217,23 +223,33 @@ function inlineFormSchema(formComponent, schemas, model) {
...
-    left: inlineFormSchemas(columnsComponent.left, schemas, model),
-    right: inlineFormSchemas(columnsComponent.right, schemas, model),
+    left: inlineReferences(columnsComponent.left, schemas, actions, model),
+    right: inlineReferences(columnsComponent.right, schemas, actions, model),
   };
 }
 
-function getSchemaById(schemas, id) {
-  return schemas.find((schema) => schema.$id === id);
+function getById(references, id) {
```
Comment (StevePole) [inline packages/dynamic-flows/src/flow/layoutService/LayoutService.js:251]: Not sure these are 'references' they are the thing being referenced...  I actually wonder if there should be a getSchemaById and a getActionById.  I appreciate the implementation is the same, but they are different things, and could diverge in future.

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js [hunk 2]
```diff
@@ -247,6 +247,27 @@ describe('Given a utility service for handling dynamic layouts', () => {
...
     ];
 
+    const actions = [
+        {
+          $id: "#submitMyDetails",
+          title: "Continue",
+          url: "/v3",
+          type: "primary",
+          method: "POST",
+          skipValidation: false,
```
Comment (StevePole) [inline packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js:257]: Is this a thing in V3?  I don't think it is...

**Review by StevePole — COMMENTED** [review]
Couple small questions

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js (hunk 2 above)
Comment (eddiewoodleytw) [reply packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js:257]: I don't think it is either, definitely not implemented on iOS client.

Comment (diego-transferwise) [reply packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js:257]: I got it from the backend library, how outdate is it? https://github.com/transferwise/dynamic-interfaces/blob/master/v3/src/main/java/com/transferwise/df/v3/steps/internal/Action.java#L30

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.js (hunk 1 above)
Comment (diego-transferwise) [reply packages/dynamic-flows/src/flow/layoutService/LayoutService.js:251]: It makes sense.

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js (hunk 2 above)
Comment (StevePole) [reply packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js:257]: In principle I think we should remove it.  The focus with JSON schema is on specifying what make a valid model, so it's a bit backwards to skip validating it.  We have 'GET' method actions, if we want to navigate without submitting a model and therefore without validation.

**Review by StevePole — APPROVED** [review]
LGTM 👍

**Pull request merged**